pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2335,6 +2343,12 @@ type CephObjectRealmList struct {
 // ObjectRealmSpec represent the spec of an ObjectRealm
 type ObjectRealmSpec struct {
 	Pull PullSpec `json:"pull,omitempty"`
+
+	// Set this realm as the default in Ceph. Only one realm should be default.
+	// This may not be set when zone is also specified; in this case, the realm
+	// referenced by the zone's zonegroup should configure defaulting behavior.
```

Comment:
If setting the default fails when deploying Reef (v18), IMO it's sufficient to document this requirement until we can fix #16211, something like this:
```suggestion
	// referenced by the zone's zonegroup should configure defaulting behavior.
	// Requires Squid (v19) or newer.
```